### PR TITLE
Use using instead of typedef [common]

### PR DESCRIPTION
**Pull request by SunBlack**
Changes are done by: `run-clang-tidy -header-filter='.' -checks='-,modernize-use-using' -fix`

### CODE REVIEW

File: common/include/pcl/common/fft/_kiss_fft_guts.h [hunk 1]
```diff
@@ -15,7 +15,7 @@ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
 /* kiss_fft.h
    defines kiss_fft_scalar as either short or a float type
    and defines
-   typedef struct { kiss_fft_scalar r; kiss_fft_scalar i; }kiss_fft_cpx; */
+   using r = kiss_fft_cpx { kiss_fft_scalar; kiss_fft_scalar i; }; */
```
Comment (taketwo) [inline common/include/pcl/common/fft/_kiss_fft_guts.h]: This is wrong I believe

File: common/include/pcl/register_point_struct.h [hunk 2]
```diff
@@ -336,13 +336,13 @@ namespace pcl
 
 // \note: the mpl::identity weirdness is to support array types without requiring the
 // user to wrap them. The basic problem is:
-// typedef float[81] type; // SYNTAX ERROR!
-// typedef float type[81]; // OK, can now use "type" as a synonym for float[81]
+// using type = float[81]; // SYNTAX ERROR!
```
Comment (taketwo) [inline common/include/pcl/register_point_struct.h]: Are these modified comments true?

File: common/include/pcl/common/fft/_kiss_fft_guts.h (hunk 1 above)
Comment (SergioRAgostinho) [reply common/include/pcl/common/fft/_kiss_fft_guts.h]: Agreed. There's no separate namespace between structs and the remaining types in C++. This feels like a remnant from C code.

File: common/include/pcl/register_point_struct.h (hunk 2 above)
Comment (SergioRAgostinho) [reply common/include/pcl/register_point_struct.h]:
```cpp
using type = float[81];
```
Compiles with `using`. The whole block comment can be removed.

File: common/include/pcl/common/fft/_kiss_fft_guts.h (hunk 1 above)
Comment (SunBlack) [reply common/include/pcl/common/fft/_kiss_fft_guts.h]: Well, seems my Regex to replace everything clang-tidy missed, should be a bit adjusted ;).

File: common/include/pcl/register_point_struct.h (hunk 2 above)
Comment (taketwo) [reply common/include/pcl/register_point_struct.h]: Only there won't be any explanation for the "`mpl::identity` weirdness" anymore. But that's fine.

**Pull request merged**